python/cudf/cudf/tests/test_column.py

```diff
@@ -79,6 +79,49 @@ def test_column_offset_and_size(pandas_input, offset, size):
     assert_eq(expect, got)
 
 
+def column_slicing_test(col, offset, size, cast_to_float=False):
+    sl = slice(offset, offset + size)
+    col_slice = col[sl]
+    series = cudf.Series(col)
+    sliced_series = cudf.Series(col_slice)
+
+    if cast_to_float:
+        pd_series = series.astype(float).to_pandas()
+        sliced_series = sliced_series.astype(float)
+    else:
+        pd_series = series.to_pandas()
+
+    if cudf.utils.dtypes.is_categorical_dtype(col.dtype):
+        # The cudf.Series is constructed from an already sliced column, whereas
+        # the pandas.Series is constructed from the unsliced series and then
+        # sliced, so the indexes should be different and we must ignore it.
+        # However, we must compare these as frames, not raw arrays,  because
+        # numpy comparison of categorical values won't work.
+        assert_eq(
+            pd_series[sl].reset_index(drop=True),
+            sliced_series.reset_index(drop=True),
+        )
+    else:
+        assert_eq(np.asarray(pd_series[sl]), sliced_series.to_array())
+
+
+@pytest.mark.parametrize("offset", [0, 1, 15])
+@pytest.mark.parametrize("size", [50, 10, 0])
+def test_column_slicing(pandas_input, offset, size):
+    col = cudf.core.column.as_column(pandas_input)
+    column_slicing_test(col, offset, size)
+
+
+@pytest.mark.parametrize("offset", [0, 1, 15])
+@pytest.mark.parametrize("size", [50, 10, 0])
+@pytest.mark.parametrize("precision", [2, 3, 5])
+@pytest.mark.parametrize("scale", [0, 1, 2])
+def test_decimal_column_slicing(offset, size, precision, scale):
+    col = cudf.core.column.as_column(pd.Series(np.random.rand(1000)))
```

Comment:
Should we make this test more deterministic by calling to numpy random seed?

Reply:
@galipremsagar I considered that. I didn't set a seed here to mimic the behavior for non-decimal inputs, which use the `pandas_input` fixture that also generates unseeded random numbers (integers in that case). There's also something to be said for intentionally allowing random inputs as a form of test fuzzing basically. 

That said, I'm fine changing this if you (or other reviewers) prefer. We should probably make it consistent with the `pandas_input` fixture either way.

Reply:
I think we can leave it as it since the randomness is contained to a limit(0 to 1).

Reply:
Historically we've set the seed I believe, just because while fuzzing casts a wide net, it needs a proper framework that can accurately capture the conditions at the time of the failure or it can be hard to determine what went wrong. @galipremsagar made a nice framework for doing exactly that for the I/O readers that we could probably adapt for other tests, but that's definitely out of scope for this.